src/main/java/com/cloudbees/jenkins/support/api/FileContent.java

```diff
@@ -97,10 +146,35 @@ protected InputStream getInputStream() throws IOException {
     }
 
     @Override
-    public long getTime()  throws IOException {
+    public long getTime() throws IOException {
         return file.lastModified();
     }
 
+    // Check if the file is binary or not
+    private boolean isBinary() {
```

Comment:
Would moving this to StreamUtils make sense?

Reply:
Not in my opinion. This method is processing something too specific.